ingestion/functions/retrieval/retrieval.py

```diff
@@ -370,27 +636,64 @@ def run_retrieval(tempdir=TEMP_PATH):
         auth_headers = common_lib.obtain_api_credentials(s3_client)
     upload_id = common_lib.create_upload_record(
         env, source_id, auth_headers, cookies)
-    url, source_format, parser, date_filter, stable_identifiers = get_source_details(
-        env, source_id, upload_id, auth_headers, cookies)
+    (url, source_format, parser, date_filter, stable_identifiers,
+     uploads_history) = get_source_details(
+         env, source_id, upload_id, auth_headers, cookies)
 
     if not stable_identifiers:
         logger.info(f"Source {source_id} does not have stable identifiers\n"
-              "Ingesting entire dataset and ignoring date filter and date ranges")
+                    "Ingesting entire dataset and ignoring date filter and date ranges")
         date_filter = {}
         parsing_date_range = {}
     url = format_source_url(url)
     file_names_s3_object_keys = retrieve_content(
-        env, source_id, upload_id, url, source_format, auth_headers, cookies, tempdir=tempdir)
-    for file_name, s3_object_key in file_names_s3_object_keys:
+        env, source_id, upload_id, url, source_format, auth_headers, cookies,
+        tempdir=tempdir, uploads_history=uploads_history)
+    file_opts = {}
+    for file_name, s3_object_key, *opts in file_names_s3_object_keys:
         upload_to_s3(file_name, s3_object_key, env,
                      source_id, upload_id, auth_headers, cookies)
+        # parse options while we're here
+        key = s3_object_key
+        file_opts[s3_object_key] = {}
+        if opts:
+            # Should the file be parsed?
+            file_opts[key]['parseit'] = (opts[0]['parseit']
```

Comment:
can be shortened to opts[0].get('parseit', True)

Reply:
👍